da/avail/avail.go

```diff
@@ -0,0 +1,227 @@
+package avail
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	ds "github.com/ipfs/go-datastore"
+	"github.com/rollkit/rollkit/da"
+	"github.com/rollkit/rollkit/da/avail/datasubmit"
+	"github.com/rollkit/rollkit/log"
+	"github.com/rollkit/rollkit/types"
+)
+
+type Config struct {
+	BaseURL    string  `json:"base_url"`
+	Seed       string  `json:"seed"`
+	ApiURL     string  `json:"api_url"`
+	AppID      int     `json:"app_id"`
+	confidence float64 `json:"confidence"`
+}
+
+type DataAvailabilityLayerClient struct {
+	namespace types.NamespaceID
+	config    Config
+	logger    log.Logger
+}
+
+type Confidence struct {
+	Block                uint32  `json:"block"`
+	Confidence           float64 `json:"confidence"`
+	SerialisedConfidence *string `json:"serialised_confidence,omitempty"`
+}
+
+type AppData struct {
+	Block      uint32   `json:"block"`
+	Extrinsics []string `json:"extrinsics"`
+}
+
+var _ da.DataAvailabilityLayerClient = &DataAvailabilityLayerClient{}
+var _ da.BlockRetriever = &DataAvailabilityLayerClient{}
+
+// Init initializes DataAvailabilityLayerClient instance.
+func (c *DataAvailabilityLayerClient) Init(namespaceID types.NamespaceID, config []byte, kvStore ds.Datastore, logger log.Logger) error {
+	c.logger = logger
+
+	if len(config) > 0 {
+		return json.Unmarshal(config, &c.config)
+	}
+
+	return nil
+}
```

Comment:
Taking namespace, config and kvStore as parameters and not at all using them in this method. And Init method not initializing any DataAvailabilityLayerClient instance.